Bugsnag/Configuration/BugsnagConfiguration.m

```diff
@@ -216,6 +215,10 @@ - (instancetype _Nonnull)initWithApiKey:(NSString *_Nonnull)apiKey
     #elif BSG_PLATFORM_OSX
         _appType = @"macOS";
     #endif
+
+    [self setAppVersion:NSBundle.mainBundle.infoDictionary[@"CFBundleShortVersionString"]];
```

Comment:
Do we definitely need to do this?

IIRC the bug was that we were reporting the wrong `bundleVersion` - isn't the removal of `_bundleVersion = NSBundle.mainBundle.infoDictionary[@"CFBundleVersion"];` enough to fix that?

Reply:
There were two problems that this change addresses:
- `CFBundleShortVersionString` wasn't being stored at all, which caused the issue where changing app version between crashing and reporting caused mismatched `CFBundleShortVersionString` and `CFBundleVersion`, reporting to a nonexistent version in the dashboard.
- Neither `CFBundleVersion` nor `CFBundleShortVersionString` were being put into `BugsnagConfiguration` internal cache because they were being stored to the member variable directly rather than using the setter. This caused them to be omitted from all events, which caused the event reader to fallback to current values for these, causing the old error to be reported to the new version if the app version changed between crashing and reporting.